### ISSUE TITLE
Support multiple "documentation pages"

### ISSUE THREAD

**Issue by suan** [ruby-grape/grape-swagger-rails#34]
Right now grape-swagger-rails expects to be mounted only once, at a single route. It also expects a single global definition of `options.url`, `options.app_url`, etc.

It would be really useful if we could mount multiple "swagger roots", just like we can mount multiple APIs at different routes using Grape itself. I think it'll also go a long way towards supporting issues like https://github.com/ruby-grape/grape-swagger/issues/141 and at the very least provide a nice workaround

Would this be something you'd be open to supporting? I'm thinking it'll look something like you can specify a set of `options` per-mount and you can mount multiple subclasses of `GrapeSwaggerRails::Engine` at different routes.

_P/S: The reason we need this now is due to Swagger 2.0's decision to only support one endpoint/operation for each (unique resource path + HTTP method) combo. The version of `swagger-ui` vendored here assumes this and only documents the last-mounted endpoint for each path. Since we use accept-header-only versioning, obviously this is very bad for us._

**Comment by dblock** [comment]
I would welcome a PR for this. Ideally we should do away with these global options if possible.

**Comment by matfiz** [comment]
@suan I had a simmilar issue. In the project, I am using `Accept-Version` versioning header. I have added the following code to the Rails initializer:

    GrapeSwaggerRails.options.before_filter do |request|
      if (version = request.query_parameters['version']).present?
        GrapeSwaggerRails.options.headers['Accept-Version'] = version
      end
    end

Then I needed to modify a bit *grape-swagger-rails* to ensure the headers are taken into consideration upon first `SwaggerUi.load()` (look at my comment here https://github.com/ruby-grape/grape-swagger-rails/pull/25). You can find my forked library here, feel free to use it: https://github.com/GeecoLABs/grape-swagger-rails

**Comment by svobom57** [comment]
+1 for this. Not having this kills the concept of versioned API (or at least the documentation part). :(

**Comment by moskvin** [comment]
The main use case behind this issue is now covered by multiple spec support in the current UI.

You can now configure:

```ruby
GrapeSwaggerRails.options.urls = [
  { name: 'v1', url: '/api/v1/swagger_doc' },
  { name: 'v2', url: '/api/v2/swagger_doc', default: true }
]
```

That does not introduce per-mount isolated option sets or multiple engine subclasses, but it does solve the practical "multiple documentation pages/specs" need inside the current architecture.

I'd close this as substantially resolved. If there is still a need for true per-mount isolated engine configuration, that should come back as a new issue with the current `urls` support as context.
